app/src/main/kotlin/com/intive/picover/parties/repository/PartiesRepository.kt

```
@@ -20,6 +21,15 @@ class PartiesRepository @Inject constructor(
 			}
 		}
 
+	suspend fun addParty(partyRemote: PartyRemote) =
+		runCatching {
+			partiesReference.push().let { databaseReference ->
+				databaseReference.key?.let { key ->
```

so what will happen now if the key is null ? just nothing will happen ? shouldn't we force the key here ?  is it possible to have no key here ?

---

@marcinWisnia According to the documentation "
The last token in the location pointed to by this reference or null if this reference points to the database root". So it is very unlikely.

---

@kmaslowiec It's not an answer for @marcinWisnia questions :)

---

Ok, I will try to reproduce that tomorrow morning

---

There is nothing to reproduce, just think about consequences :)

---

@MarekMacko @marcinWisnia If something happens with the Firebase Server, the part databaseReference.setValue(partyRemote.copy(id = key)).await() will not be called therefore, the Party will not be added. Unfortunately, nobody will know about it. I could generate a random number for this scenario but I would rather throw an Exception and later handle it with the SnackBar with an error message.

---

@kmaslowiec Perfectly, that's the point and right conclusion :)

---

@kmaslowiec that's the spirit :D

---

Updated. I was wondering if I should throw more specific error such as IllegalStateExcepption but I think a general error message 'Something went wrong" should be enough for such a corner case.